src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's contract in the address book.
+ */
+
+public class Contact {
+	
+    public final String value;
+    private boolean isPrivate;
+    
+	public Contact(String value, boolean isPrivate) {
+		this.value = value;
+		this.isPrivate = isPrivate;
+	}
+
+    @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
by virtue of inheritance, email and address may be validated as the same instance, no?
it is better to leave out the `equals` method in the child class